client/add.go

```diff
@@ -2,16 +2,25 @@ package client
 
 import (
 	"fmt"
-	"io/ioutil"
-	"os"
+
+	"github.com/pinterest/knox"
 )
 
+func init() {
+	cmdAdd.Run = runAdd // break init cycle
+}
+
 var cmdAdd = &Command{
-	Run:       runAdd,
-	UsageLine: "add <key_identifier>",
+	UsageLine: "add [--key-template template_name] <key_identifier>",
 	Short:     "adds a new key version to knox",
 	Long: `
-add adds a new key version to an existing key in knox. Key data should be sent to stdin.
+Add will add a new key version to an existing key in knox. Key data of new version should be sent to stdin unless a key-template is specified.
+
+First way: key data of new version is sent to stdin.
+Please run "knox add <key_identifier>". 
+
+Second way: the key-template option can be used to specify a template to generate the new key version, instead of stdin. For available key templates, run "knox key-templates".
+Please run "knox add --key-template template_name <key_identifier>".
```

Comment:
nit: <template_name> to indicate its a user specified variable (same as <key_identifier>.

Reply:
Thank you for your comment! I revised the code accordingly.